tests/driver/api.test.ts

```diff
@@ -0,0 +1,633 @@
+        return mongooseInstance;
+    }
+
+    async function createClientsAndModels(isAstra: boolean) {
+        let Product: Model<any>, Cart: Model<any>, astraMongoose: Mongoose | null = null, jsonAPIMongoose: Mongoose | null = null;
+        const productSchema = new mongoose.Schema({
+            name: String,
+            price: Number,
+            expiryDate: Date,
+            isCertified: Boolean,
+            category: String
+        });
+        const cartSchema = new mongoose.Schema({
+            name: String,
+            products: [{ type: Schema.Types.ObjectId, ref: 'Product' }]
+        });
+        if (isAstra) {
+            astraMongoose = await getInstance();
+            Product = astraMongoose.model('Product', productSchema);
+            Cart = astraMongoose.model('Cart', cartSchema);
+            // @ts-ignore - these are config options supported by stargate-mongoose but not mongoose
+            await astraMongoose.connect(dbUri, {isAstra: true, logSkippedOptions: true});
+            await Promise.all(Object.values(astraMongoose.connection.models).map(Model => Model.init()));
+        } else {
+            // @ts-ignore
+            jsonAPIMongoose = await getInstance();
+            Product = jsonAPIMongoose.model('Product', productSchema);
+            Cart = jsonAPIMongoose.model('Cart', cartSchema);
+            const options = {
+                username: process.env.STARGATE_USERNAME,
+                password: process.env.STARGATE_PASSWORD,
+                authUrl: process.env.STARGATE_AUTH_URL,
+                logSkippedOptions: true
+            };
+            // @ts-ignore - these are config options supported by stargate-mongoose but not mongoose
+            await jsonAPIMongoose.connect(dbUri, options);
+            await Promise.all(Object.values(jsonAPIMongoose.connection.models).map(Model => Model.init()));
+        }
+        mongooseInstance = isAstra ? astraMongoose : jsonAPIMongoose;
+        return {Product, Cart, astraMongoose, jsonAPIMongoose};
+    }
+
+    async function dropCollections(isAstra: boolean, astraMongoose: mongoose.Mongoose | null, jsonAPIMongoose: mongoose.Mongoose | null, collectionName: string) {
+        if (isAstra) {
+            astraMongoose?.connection.dropCollection(collectionName);
+        } else {
+            jsonAPIMongoose?.connection.dropCollection(collectionName);
+        }
+    }
+
+    describe('Options & Data type tests', () => {
+        it('test strict and strictQuery options', async () => {
+            //strict is true, so extraCol should not be saved
+            const saveResponseWithStrictTrue = await new Product({
+                name: 'Product 1',
+                price: 10,
+                isCertified: true,
+                category: 'cat 1',
+                extraCol: 'extra val'
+            }, null, {
+                strict: true
+            }).save();
+            assert.strictEqual(saveResponseWithStrictTrue.name, 'Product 1');
```

Comment:
2 details worth noting:

1. `await doc.save()` returns `doc`, so you're not really testing anything here. I'd recommend querying the document from the database and asserting that has the correct properties.
2. `strict` is true by default, so no need to set `strict: true`.